superset/common/query_actions.py

```diff
@@ -101,14 +107,28 @@ def _get_full(
     datasource = _get_datasource(query_context, query_obj)
     result_type = query_obj.result_type or query_context.result_type
     payload = query_context.get_df_payload(query_obj, force_cached=force_cached)
+    presigned_output_location = None
+
+    if datasource.database.backend == "awsathena" and "presto" in datasource.database.name.lower() \
```

Comment:
We should move this logic to the DB engine specs themselves, to abstract the feature.